File: labs/architecture-examples/angularjs_saltarelle/cs/directives/TodoBlur.cs
using System;
using jQueryApi;
using ng;

namespace todo
{
    public class TodoBlur
    {
        public Action<IScope, jQueryObject, TodoAttrs> link;
        private readonly ILogService log;

        public TodoBlur(ILogService log)
        {
            this.log = log;
            link = linkFn;
        }

        private void linkFn(IScope scope, jQueryObject elem, TodoAttrs attrs)
        {
            log.log("TodoBlur link");
            elem.Bind("blur", e => {
                log.log("TodoBlur bind");
                scope.Apply(attrs.todoBlur);
            });
        }
    }
}

File: labs/architecture-examples/angularjs_typescript/js/directives/TodoBlur.ts
/// <reference path='../libs/angular-1.0.d.ts' />
'use strict';


/**
 * Directive that executes an expression when the element it is applied to loses focus.
 */
class TodoBlur {
    public link: ($scope: ng.IScope, elem: JQuery, attrs: any) => any;

    constructor() {
        this.link = (s, e, a) => this.linkFn(s, e, a);
    }

    linkFn($scope: ng.IScope, elem: JQuery, attrs: any): any {
        elem.bind('blur', () => {
            $scope.$apply(attrs.todoBlur);
        });
    };
}
